lib/src/install.rs

```diff
@@ -159,6 +157,18 @@ pub(crate) struct InstallConfigOpts {
     #[clap(long)]
     #[serde(default)]
     pub(crate) generic_image: bool,
+
+    /// The stateroot name to use. Defaults to `default`.
+    #[clap(long, default_value = ostree_ext::container::deploy::STATEROOT_DEFAULT, hide = true)]
+    #[serde(default = "default_stateroot")]
+    pub(crate) stateroot: String,
+}
+
+// TODO: Is this necessary? Added so that the install_opts_serializable test passes, because that
+// test is doing deserialization, but will users ever really do deserialization themselves, and if
+// yes, should we help them by providing a default value?
```

Comment:
Users won't deserialize, but the idea is we may do so internally as part of things like https://github.com/containers/bootc/pull/78 - that's why the serialization was added.

Basically for various reasons we end up wanting to re-exec the current process, and it seems more elegant to do so via `vec<string> -> clap -> serde -> json` instead of `vec<string> -> clap -> vec<string>` (or trickier, trying to skip back over what clap did and find the original `vec<string>`).

It may be simpler on the implementation to keep it as an `Option<String>`, then serde and clap will handle it the same way, and we do the `unwrap_or_else()` in the install code.

Reply:
>Basically for various reasons we end up wanting to re-exec the current process, and it seems more elegant to do so via vec<string> -> clap -> serde -> json instead of vec<string> -> clap -> vec<string> (or trickier, trying to skip back over what clap did and find the original vec<string>).

In that case, iiuc in the re-exec scenario we still don't need to provide a serde default for deserialization purposes because no user will hand-craft the serialized form for us to deserialize. We'll always be serializing clap-initialized structs, so only clap really needs to have a default. The clap default ensures all structs serialized and deserialized will already have this value initialized

But I've switched to Option<String> regardless to keep it simple